### PR TITLE
Add documentation fixes for library type_context.lean and format.lean

### PR DESCRIPTION
**Pull request by EdAyers**
Various fixes for docstrings

- Resolved `[todo]`s in `init/meta/format.lean`
- change old references to `tco` to `type_context`.
- add clarification of `type_context.get_context`
- add clarification of `type_context.level.assign`.
- add docstring for `type_context.run`

### CODE REVIEW

**Comment by cipher1024** [comment]
I appreciate the documentation! I had some complaints about indenting comments in docstrings. People seem to prefer

```
/-- foo foo bar foo
new line foo bar foo
-/
```

because of how VSCode reads and displays the comments in hover text.

**Comment by gebner** [comment]
This is a definite improvement.  Do you plan to fix the other `[TODO]`s as well?

**Comment by EdAyers** [comment]
Yes, I've just had a look through them now. Some of them were just notes to myself to understand some obscure config flag. I can't understand what some of the config flags do still so is it ok to just remove the '[todo]'?

**Comment by gebner** [comment]
Yes, please remove the TODOs.  If I (or somebody else) knows what the flags do, we can always add it as a suggestion.

**Comment by EdAyers** [comment]
done, see the commit message of a6bf1c8 for more info.

File: library/init/meta/environment.lean [hunk 1]
```diff
@@ -12,17 +12,15 @@ meta constant environment : Type
...
 This helps the simplifier / rewriter not have to expand projectors.
 Eg `a (mk x y)` will automatically reduce to `x`.
 If you `extend` a structure, all of the projections on the parent will also be created for the child.
+Projections are also treated differently in the VM for efficiency.
 
-[TODO] any other reasons Lean treats projections differently to regular declarations?
-    I know that you get projection macros.
-[TODO] Are there projections that aren't arguments to constructors?
-[NOTE] projectors have nothing to do with the dot `mylist.map` syntax.
+Note that projectors have nothing to do with the dot `mylist.map` syntax.
```
Comment (gebner) [inline library/init/meta/environment.lean:23]:
```suggestion
Note that projections have nothing to do with the dot `mylist.map` syntax.
```

**Pull request merged**